controllers/managedfusionoffering_controller.go

```diff
@@ -190,6 +194,15 @@ func (r *ManagedFusionOfferingReconciler) reconcilePhases() (reconcile.Result, e
 
 }
 
+func validateOfferingConfig(offering *v1alpha1.ManagedFusionOffering) error {
+
+	var offeringConfigYaml interface{}
+	if err := yaml.Unmarshal([]byte(offering.Spec.Config), &offeringConfigYaml); err != nil {
+		return fmt.Errorf("invalid yaml found in ManagedFusionOffering config: %v", err)
+	}
+	return nil
+}
```

Comment:
Why do we need to validate here and then unmarshal again in the offering code? 
Can we just unmarshal once just before pluginReconcile? 

The way I would do that is by asking the offering code to provide an empty instance of the offeringConfig struct as an interface. Unmarshl into that interface, using `StrictUnmarshal` (this will cover both YAML validity check and specific offering validity check) then pass that into to the offering reconcile code. 

The offering can then use `config.(OfferingConfigStructType)` to case it into the correct type and use that

Reply:
@nb-ohad I tried unmarshalling it into a interface and type cast it using config.(OfferingConfigStructType), it's throwing a error.

We can unmarshall into a interface or map[sting]interface{} and pass that into offering reconcile code and then parseSpec can use that interface to read values